core/src/test/java/com/linecorp/armeria/server/cors/HttpServerCorsTest.java

```diff
@@ -609,4 +652,92 @@ public void testNoCorsWithPartialBinding() {
+                .isEqualTo("http://example.com");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_METHODS)).isEqualTo("GET");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_HEADERS)).isNull();
+    }
+
+    @Test
+    public void testAnnotatedServiceAllowAllHeaders() {
+        final WebClient client = client();
+        final HttpRequest preflightReq = HttpRequest.of(
+                RequestHeaders.of(HttpMethod.OPTIONS, "/cors14/index",
+                                  HttpHeaderNames.ORIGIN, "http://example.com",
+                                  HttpHeaderNames.ACCESS_CONTROL_REQUEST_METHOD, "GET",
+                                  HttpHeaderNames.ACCESS_CONTROL_REQUEST_HEADERS, "foo,bar"));
+        final AggregatedHttpResponse res = client.execute(preflightReq).aggregate().join();
+        assertThat(res.status()).isEqualTo(HttpStatus.OK);
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_ORIGIN))
+                .isEqualTo("http://example.com");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_METHODS)).isEqualTo("GET");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_HEADERS)).isEqualTo("foo,bar");
+    }
+
+    @Test
+    public void testAllowAllHeadersUsingWildcard() {
+        final WebClient client = client();
+
+        HttpRequest preflightReq = HttpRequest.of(
+                RequestHeaders.of(HttpMethod.OPTIONS, "/cors15",
+                                  HttpHeaderNames.ORIGIN, "http://example.com",
+                                  HttpHeaderNames.ACCESS_CONTROL_REQUEST_METHOD, "GET",
+                                  HttpHeaderNames.ACCESS_CONTROL_REQUEST_HEADERS, "foo,bar"));
+        AggregatedHttpResponse res = client.execute(preflightReq).aggregate().join();
+        assertThat(res.status()).isEqualTo(HttpStatus.OK);
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_ORIGIN))
+                .isEqualTo("http://example.com");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_METHODS)).isEqualTo("GET");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_HEADERS)).isEqualTo("*");
+
+        preflightReq = HttpRequest.of(
+                RequestHeaders.of(HttpMethod.OPTIONS, "/cors15",
+                                  HttpHeaderNames.ORIGIN, "http://example.com",
+                                  HttpHeaderNames.ACCESS_CONTROL_REQUEST_METHOD, "GET"));
+        res = client.execute(preflightReq).aggregate().join();
+        assertThat(res.status()).isEqualTo(HttpStatus.OK);
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_ORIGIN))
+                .isEqualTo("http://example.com");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_METHODS)).isEqualTo("GET");
+        assertThat(res.headers().get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_HEADERS)).isNull();
```

Comment:
If use have used `allowRequestHeaders(*)`, behavior will be changed.
Server sets `Access-Control-Allow-Headers` to `*` every time even request doesn't have `Access-Control-Request-Headers`. But this PR will omit `Access-Control-Allow-Headers` if the request doesn't have.
Could you check it? Is it may breaking change?